tests/js/spec/components/loading/loadingContainer.spec.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import {render} from '@testing-library/react';
+
+import LoadingContainer from 'app/components/loading/loadingContainer';
+
+const loadingContainerChildren = 'hello!';
+
+function renderComponent({
+  children = loadingContainerChildren,
+  ...props
+}: Partial<React.ComponentProps<typeof LoadingContainer>>) {
+  const utils = render(
+    <LoadingContainer {...props}>
+      <div>{children}</div>
+    </LoadingContainer>
+  );
+  return {...utils};
+}
+
+// TODO(Priscila): check the loading state tests
+describe('LoadingContainer', () => {
+  it('default state', () => {
+    const {getByText, queryByTestId} = renderComponent({});
+    expect(getByText(loadingContainerChildren)).toBeTruthy();
+    expect(queryByTestId('loading-indicator')).toBeNull();
+  });
+
+  it('handles loading state - with children', () => {
+    const {getByText, getByTestId} = renderComponent({
+      isLoading: true,
+    });
+    expect(getByText(loadingContainerChildren)).toBeTruthy();
+    expect(getByTestId('loading-indicator')).toBeTruthy();
+  });
+
+  it('handles loading state - without children', () => {
+    const {queryByText, getByTestId} = renderComponent({
+      children: null,
+      isLoading: true,
+    });
+    expect(queryByText(loadingContainerChildren)).toBeNull();
+    expect(getByTestId('loading-indicator')).toBeTruthy();
+  });
+
+  it('reloading state - with children', () => {
+    const {getByText, getByTestId} = renderComponent({
+      isReloading: true,
+    });
```

Comment:
Is there anything in react-testing that is analogous to `wrapper.update()`? We often resort to having to use that to ensure reflux updates have been flushed out.

Reply:
Yes, I would be curious to see a test in `issueList/overview` converted because those are probably more complex set of tests and are also the slowest.

Reply:
@billyvg I will convert some of those tests 😉
@markstory I don't know.. I will do some research and let you know

Reply:
@markstory I did some research and I could not find anything similar... There is the rerender function but it's used only if you want to test updated props. https://testing-library.com/docs/react-testing-library/api#rerender

Reply:
@markstory 

According to the enzyme docs, `.update` is useful to run before checking the render output if something external may be updating the state of the component somewhere... 

After doing some more research and finally fully understanding what` .update` does,  I truly believe that we can use `waitFor` to replace `.update()` from enzyme. Please see:

**waitFor**
> This can be useful if you have a unit test that mocks API calls and you need to wait for your mock promises to all resolve.

https://testing-library.com/docs/dom-testing-library/api-async#waitfor

Reply:
Wow `issueList/overview.spec` has `1664` lines of code 😬 I'm not sure I can finish converting it today before our meeting